src/features/Chart/ChartSlice.ts

```diff
@@ -1,59 +1,51 @@
-import { SagaIterator } from '@redux-saga/types';
-import moment from 'moment';
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
-import { getContext, call, put, takeEvery } from 'redux-saga/effects';
-import { ChartDataPoint } from './ChartManager';
-import { RootState } from './../../common/store/store';
-import { DEPENDENCY, REDUCER } from '../../common/constants';
-
-interface ChartSliceState {
-  listLoadingById: Record<string, boolean>;
-  listDataById: Record<string, ChartDataPoint[]>;
-  detailDataPoints: ChartDataPoint[];
-}
+import { SagaIterator } from '@redux-saga/types';
```

Comment:
main changes here are extracting the types out to a type file (sorry) and then to have the `hydrate` action which fetches all the chart data in a single query.

Other changes are mainly better typing